### ISSUE TITLE
Different IS NULL behavior for ad-hoc vs compiled

### ISSUE THREAD

**Issue by mtornwall** [Starcounter/Starcounter.Linq#62]
In Starcounter.Linq, there's a subtle difference in behavior between ad-hoc queries and compiled queries. Consider these two cases:

```
Node parent = null;
var results = DbLinq.Objects<Node>()
    .Where(o => o.Parent == parent).ToList();
// results contains all nodes whose parent is null,
// because Starcounter.Linq generated an IS NULL condition!
```

... and:

```
Node parent = null;
var query = DbLinq.CompileQuery((Node parent) =>
    DbLinq.Objects<Node>().Where(o => o.Parent == parent));
var results = query(parent);
// results contains no nodes at all, because
// Starcounter.Linq generated a regular equality condition,
// which is false for null!
```

Not sure if this is a bug or feature.

**Comment by un-tone** [comment]
The behavior is described in the README. Please look at https://github.com/Starcounter/Starcounter.Linq/tree/develop#null-values. This is how it works, I don't see another solution here, only the one described in the README.

**Comment by un-tone** [comment]
From [comment](https://github.com/Starcounter/Starcounter.Linq/pull/66#issuecomment-439995817):

Now I think what if for compiled query we'll prepare more than one translated SQL queries? For example, for

```cs
DbLinq.CompileQuery((Foo compareAgainst) =>
    DbLinq.Objects<Foo>().Where(o => o == compareAgainst && o.Name == name)
)
```

it will be:

```sql
SELECT F FROM "Foo" F WHERE F = ? AND F."Name" = ?
SELECT F FROM "Foo" F WHERE F = IS NULL AND F."Name" = ?
SELECT F FROM "Foo" F WHERE F = ? AND F."Name" IS NULL
SELECT F FROM "Foo" F WHERE F = IS NULL AND F."Name" IS NULL
```
And on executing it will check which parameters is null and select most appropriate SQL query.

**Comment by mtornwall** [comment]
> Now I think what if for compiled query we'll prepare more than one translated SQL queries? For example, for

This would grow combinatorially. Not realistic.

**Comment by un-tone** [comment]
Published the updated packages (1.3.1, 2.3.1, 3.3.1).

**Comment by un-tone** [comment]
Closing this for now. Fixed in https://github.com/Starcounter/Starcounter.Linq/pull/66
